packages/application/src/shell.ts

```diff
@@ -743,6 +743,9 @@ export class LabShell extends Widget implements JupyterFrontEnd.IShell {
     if (options.ref) {
       ref = find(dock.widgets(), value => value.id === options.ref!) || null;
     }
+
+    widget.title.dataset = { ...widget.title.dataset, id: widget.id };
```

Comment:
I think it would be useful to add a comment here about why we add this dataset item.

Reply:
Done!